### DISCUSSION TITLE
Controlling prefix sh: in SBDH xml elements

### DISCUSSION THREAD

**Discussion by binaradarsha** [phax/phase4#136] [General]
Hi,

In Peppol world it requires SBDH documents. As in Peppol documentation examples <StandardBusinessDocument>, <StandardBusinessDocumentHeader> etc. elements don't have the prefix `sh:`. But if we input an Invoice to the Phase4PeppolSender.Builder the output SBDH doc is generated with prefix `sh:`. 

So are the both ways (with and without prefix `sh:`) totally accepted in Peppol, and in general?
Does Phase4PeppolSender.Builder have any method to control the presence of this prefix?

**Comment by phax** [comment]
Hi, the XML namespace prefix should in realty not matter.
The namespace prefix is a "shortcut" to avoid to always write the long namespace URI in all elements.
Take this example:
```xml
<sh:StandardBusinessDocument xmlns:sh="http://www.unece.org/cefact/namespaces/StandardBusinessDocumentHeader">
...
</sh:StandardBusinessDocument>
```
tells the XML processor: replace the prefix `sh` with `http://www.unece.org/cefact/namespaces/StandardBusinessDocumentHeader` when reading the XML.

In memory, the relevant element representation is only the combination of namespace URI `http://www.unece.org/cefact/namespaces/StandardBusinessDocumentHeader` and element name `StandardBusinessDocument` - the prefix is irrelevant for execution.

So the above example is totally equivalent to
```xml
<hugo:StandardBusinessDocument xmlns:hugo="http://www.unece.org/cefact/namespaces/StandardBusinessDocumentHeader">
...
</hugo:StandardBusinessDocument>
```

And in case you want to be very brief, you fallback to the "default prefix" (the empty string), it looks like this:
```xml
<StandardBusinessDocument xmlns="http://www.unece.org/cefact/namespaces/StandardBusinessDocumentHeader">
...
</StandardBusinessDocument>
```
It is all the same...
So the prefix does NOT matter.

Of course you can have bad implementations that rely on a certain prefix, but that would be a processing error on their side and not XML compliant.

hth

**Reply by binaradarsha** [reply]
Thank you very much for the explanation Philip!
